File: NHSOnline.Backend.Worker.Mocking/Emis/Models/BadRequestResponse.cs
using System.Collections.Generic;

namespace NHSOnline.Backend.Worker.Mocking.Emis.Models
{
    public class BadRequestResponse
    {
        public string Message { get; }

        public Dictionary<string, IEnumerable<string>> ModelState { get; }

        public BadRequestResponse(string message)
        {
            Message = message;
        }

        public BadRequestResponse(string message, string fieldName) :this(message)
        {
            ModelState = new Dictionary<string, IEnumerable<string>>
            {
                [$"request.{fieldName}"] = new[] { "An error has occurred." }
            };
        }
    }
}

File: bddtests/src/main/kotlin/mocking/emis/models/BadRequestResponse.kt
package mocking.emis.models

class BadRequestResponse(val message: String) {

    var modelState: Map<String, ArrayList<String>>? = null

    constructor(message: String, fieldName: String) : this(message) {
        modelState = mapOf(String.format("request.%1\$s", fieldName) to arrayListOf("An error has occurred."))
    }
}
